llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -643,8 +644,34 @@ bool RISCVVLOptimizer::isCandidate(const MachineInstr &MI) const {
 
   unsigned VLOpNum = RISCVII::getVLOpNum(Desc);
   const MachineOperand &VLOp = MI.getOperand(VLOpNum);
-  if (!VLOp.isImm() || VLOp.getImm() != RISCV::VLMaxSentinel)
+  if (((VLOp.isImm() && VLOp.getImm() != RISCV::VLMaxSentinel) ||
```

Comment:
Should this be
```suggestion
  if (((VLOp.isImm() && VLOp.getImm() == RISCV::VLMaxSentinel) ||
```

Reply:
I think it is supposed to be `!=`. If we are already using VLMAX, then I don't think we care about whether its TA/MU since there are no tail elements that need to be preserved.

Reply:
Ah sorry that makes sense. Just to check then, there isn't any difference between
`!VLOp.isImm() || VLOp.getImm() != RISCV::VLMaxSentinel` and `((VLOp.isImm() && VLOp.getImm() != RISCV::VLMaxSentinel) || VLOp.isReg())`?